src/Phinx/Db/Adapter/PdoAdapter.php

```diff
@@ -76,9 +76,20 @@ protected function verboseLog($message)
      */
     protected function createPdoConnection($dsn, $username = null, $password = null, array $options = [])
     {
+        $adapterOptions = $this->getOptions();
+
+        if (!isset($adapterOptions['attr_errmode'])) {
+            $adapterOptions['attr_errmode'] = PDO::ERRMODE_EXCEPTION;
+        }
+
         try {
             $db = new PDO($dsn, $username, $password, $options);
-            $db->setAttribute(PDO::ATTR_ERRMODE, PDO::ERRMODE_EXCEPTION);
+
+            foreach ($adapterOptions as $key => $option) {
+                if (strpos($key, 'attr_') === 0) {
+                    $db->setAttribute(constant('\PDO::' . strtoupper($key)), $option);
```

Comment:
We check if the constant exists and throw a useful exception if not.

Reply:
It might be easier to simply check if any of the options are valid constants and set them if they are instead of checking for attr_ and validating it.

Reply:
I still need to make a check on the string prefix, to avoid including `mysql_attr_` and `sqlsrv_attr_` values here, which are considered valid constants, but should not be set within this context.

This is also the same set-up as we use for assigning the `mysql_attr_` and `sqlsrv_attr_` variables within their respective adapters, so you would like to see said exception thrown be added there as well?

Reply:
I think an exception here is better than a php error.

Reply:
Done